### ISSUE TITLE
[Feat]  Fix extra compiler warnings

### ISSUE THREAD

**Issue by nilason** [OSGeo/grass#2747]
Enabling `-Wextra` compiler flag in combination with `-Wall` causes a number of warnings of following types:

- [ ] -Wdeprecated-non-prototype #2727 #2798
- [x] -Wempty-body #2755
- [x] -Wextra #2754
- [x] -Wignored-qualifiers #2753
- [x] -Wimplicit-fallthrough= #2760
- [x] -Wliteral-range #2748
- [x] -Wmissing-field-initializers #2753
- [x] -Wsign-compare #2761
- [x] -Wsometimes-uninitialized #2751
- [x] -Wtype-limits #2756 #2799
- [x] -Wunused-but-set-variable #2752
- [x] -Wunused-parameter #2770

with GCC 9.4 and Clang 16.0.0

The issues are of various degree of seriousness, but addressing them would improve if only a little bit on code quality and keeping the code in good condition.
This is a follow-up to #2156.

**Comment by nilason** [comment]
The GCC (gcc 11.3.0 + glibc 2.35) and Mac (clang 14.0.6 + MacOSX13.sdk) CI runners now compiles with
`-Wall -Wextra -Werror`.

There remain two code parts with `-Wdeprecated-non-prototype` warnings, which are issued with clang 16 (still not released). Those are non-trivial and I have prepared fixes (but not yet PRs). They are not urgent fixes and may be bumped to version 8.4.

Closing this as implemented.
